---a/net/minecraft/world/gen/structure/MapGenStructureIO.java
+++b/net/minecraft/world/gen/structure/MapGenStructureIO.java
@@ -1,13 +1,13 @@
 package net.minecraft.world.gen.structure;
 
-import com.google.common.collect.Maps;
 import java.util.Map;
+
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.world.World;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-public class MapGenStructureIO
-{
+import com.google.common.collect.Maps;
+
+public class MapGenStructureIO {
+
     private static final Logger logger = LogManager.getLogger();
     private static Map field_143040_a = Maps.newHashMap();
     private static Map field_143038_b = Maps.newHashMap();
@@ -16,92 +16,71 @@
     private static Map field_143037_d = Maps.newHashMap();
     private static final String __OBFID = "CL_00000509";
 
-    private static void registerStructure(Class p_143034_0_, String p_143034_1_)
-    {
+    public static void registerStructure(Class p_143034_0_, String p_143034_1_) {
         field_143040_a.put(p_143034_1_, p_143034_0_);
         field_143038_b.put(p_143034_0_, p_143034_1_);
     }
 
-    static void registerStructureComponent(Class p_143031_0_, String p_143031_1_)
-    {
+    public static void registerStructureComponent(Class p_143031_0_, String p_143031_1_) {
         field_143039_c.put(p_143031_1_, p_143031_0_);
         field_143037_d.put(p_143031_0_, p_143031_1_);
     }
 
-    public static String func_143033_a(StructureStart p_143033_0_)
-    {
-        return (String)field_143038_b.get(p_143033_0_.getClass());
+    public static String func_143033_a(StructureStart p_143033_0_) {
+        return (String) field_143038_b.get(p_143033_0_.getClass());
     }
 
-    public static String func_143036_a(StructureComponent p_143036_0_)
-    {
-        return (String)field_143037_d.get(p_143036_0_.getClass());
+    public static String func_143036_a(StructureComponent p_143036_0_) {
+        return (String) field_143037_d.get(p_143036_0_.getClass());
     }
 
-    public static StructureStart func_143035_a(NBTTagCompound p_143035_0_, World worldIn)
-    {
+    public static StructureStart func_143035_a(NBTTagCompound p_143035_0_, World worldIn) {
         StructureStart structurestart = null;
 
-        try
-        {
-            Class oclass = (Class)field_143040_a.get(p_143035_0_.getString("id"));
+        try {
+            Class oclass = (Class) field_143040_a.get(p_143035_0_.getString("id"));
 
-            if (oclass != null)
-            {
-                structurestart = (StructureStart)oclass.newInstance();
+            if (oclass != null) {
+                structurestart = (StructureStart) oclass.newInstance();
             }
-        }
-        catch (Exception exception)
-        {
+        } catch (Exception exception) {
             logger.warn("Failed Start with id " + p_143035_0_.getString("id"));
             exception.printStackTrace();
         }
 
-        if (structurestart != null)
-        {
+        if (structurestart != null) {
             structurestart.func_143020_a(worldIn, p_143035_0_);
-        }
-        else
-        {
+        } else {
             logger.warn("Skipping Structure with id " + p_143035_0_.getString("id"));
         }
 
         return structurestart;
     }
 
-    public static StructureComponent func_143032_b(NBTTagCompound p_143032_0_, World worldIn)
-    {
+    public static StructureComponent func_143032_b(NBTTagCompound p_143032_0_, World worldIn) {
         StructureComponent structurecomponent = null;
 
-        try
-        {
-            Class oclass = (Class)field_143039_c.get(p_143032_0_.getString("id"));
+        try {
+            Class oclass = (Class) field_143039_c.get(p_143032_0_.getString("id"));
 
-            if (oclass != null)
-            {
-                structurecomponent = (StructureComponent)oclass.newInstance();
+            if (oclass != null) {
+                structurecomponent = (StructureComponent) oclass.newInstance();
             }
-        }
-        catch (Exception exception)
-        {
+        } catch (Exception exception) {
             logger.warn("Failed Piece with id " + p_143032_0_.getString("id"));
             exception.printStackTrace();
         }
 
-        if (structurecomponent != null)
-        {
+        if (structurecomponent != null) {
             structurecomponent.func_143009_a(worldIn, p_143032_0_);
-        }
-        else
-        {
+        } else {
             logger.warn("Skipping Piece with id " + p_143032_0_.getString("id"));
         }
 
         return structurecomponent;
     }
 
-    static
-    {
+    static {
         registerStructure(StructureMineshaftStart.class, "Mineshaft");
         registerStructure(MapGenVillage.Start.class, "Village");
         registerStructure(MapGenNetherBridge.Start.class, "Fortress");
@@ -115,4 +115,4 @@
         ComponentScatteredFeaturePieces.registerScatteredFeaturePieces();
         StructureOceanMonumentPieces.func_175970_a();
     }
-}
+}
